src/server/Group.cpp

```diff
@@ -16,32 +18,40 @@ Group::~Group() {
 void Group::update() {
     refresh();
 
-    // Update velocity
-    sf::Vector2f new_velocity = sf::Vector2f(0.f, 0.f);
-    for(auto member: mMembers) {
-        new_velocity += member->getDirection();
-    }
-    mCircle->setVelocity(new_velocity);
+    if (isActive()) {
+        // Update velocity
+        sf::Vector2f new_velocity = std::accumulate(
+            mMembers.begin(),
+            mMembers.end(),
+            sf::Vector2f(0.f, 0.f),
+            [](sf::Vector2f curr_vel, std::shared_ptr<Player> player) {
+                return curr_vel + player->getDirection();
+            }
+        );
+        mCircle->setVelocity(new_velocity);
+
+        // Update position
+        mCircle->move();
 
-    // Update position
-    mCircle->move();
+    }
 }
 
 /**
     Sets group to active if any of its members are active.
 */
 void Group::refresh() {
-    for(auto member: mMembers) {
-        if (member->isActive()) {
-            setActive(true);
-            return;
-        }
-    }
-    setActive(false);
+    bool any_active_members = std::any_of(
+        mMembers.begin(), mMembers.end(),
+        [](std::shared_ptr<Player> player){return player->isActive();}
+    );
+    if (any_active_members)
```

Comment:
add the curly braces here, not having them is bad style

Reply:
ok